snapdtool/tool_linux.go

```diff
@@ -217,6 +217,19 @@ func IsReexecd() (bool, error) {
 	return false, nil
 }
 
+// IsFromSnapdSnap returns true when the current process binary is running from
+// the snapd snap.
+func IsFromSnapdSnap() (bool, error) {
+	exe, err := osReadlink(selfExe)
+	if err != nil {
+		return false, err
+	}
+	if !strings.HasPrefix(exe, filepath.Join(dirs.SnapMountDir, "snapd")) {
```

Comment:
does this work when we seed snapd starting from the trampoline, in that case on core18 snapd comes from a tmp directory?

Reply:
code is here:

https://github.com/snapcore/core18/blob/master/static/usr/lib/core18/run-snapd-from-snap

Reply:
Perhaps rather than checking the path, it would be better to check the mounted filesystem and major:minor that match to a loopback device?

Reply:
core20:  https://github.com/snapcore/core20/blob/master/static/usr/lib/core/run-snapd-from-snap